File: BehaviorTreesLib/BehaviorTree.cpp
#include "BehaviorTree.h"

namespace fluentBehaviorTree
{
	BehaviorTree::BehaviorTree(std::string name, Node* root)
	{
		this->setName(name);
		mRoot = root;
	}

	BehaviorTree::BehaviorTree(BehaviorTreeBuilder btb)
	{
		this->setName(btb.getName());
		mRoot = btb.getRoot();
	}

	EStatus BehaviorTree::tick()
	{
		return mRoot->tick();	
	}
}

File: C#/BehaviorTree/BehaviorTree.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace FluentBehaviorTree
{
    public class BehaviorTree
    {
        private string name;

        public string GetName() { return name; }
        public Node GetRoot() { return root; }
        protected void SetName(string newName) { name = newName; }


        private Node root;

        public BehaviorTree(string newName, Node newRoot)
        {
            name = newName;
            root = newRoot;
        }

        /// <summary>
        /// Tick tree to navigate and get result
        /// </summary>
        /// <returns></returns>
        public Status Tick()
        {
            var status = root.Tick();
            return status;
        }
    }
}
